app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarController.kt

```diff
@@ -131,6 +138,19 @@ class DefaultBrowserToolbarController(
                 item.isChecked,
                 currentSession
             )
+            ToolbarMenu.Item.AddToFirefoxHome -> {
+                ioScope.launch {
```

Comment:
Instead of making a new scope you can tell the lifecycle scope to launch on the IO dispatcher.
```suggestion
                scope.launch(IO) {
```

Reply:
We hold onto a reference of the inner scope so that we can override this with the TestCoroutineScope to ensure sequential execution. If we didn't have this, our tests would fail intermittently due to the async nature of coroutine scheduling.

I will add this as a comment for ioScope.